### DISCUSSION TITLE
How to expose a virtual machine to the public for remote access?

### DISCUSSION THREAD

**Discussion by Bazimek** [cockpit-project/cockpit-machines#571] [Q&A]
Hello I was trying everything what I read on the internet, but no luck. How can I expose a virtual machine to the public for remote access?

**Comment by KKoukiou** [comment]
@Bazimek In order that we help you, we need some concrete data for reproducing your issue. In specific:

* I have tried everything and it does not work. -> Did you follow some specific tutorial and it did not work? If yes what exactly did you try? 

The most common way of achieving the setup you want, is to create a bridge and use that in a bridge type interface in your VMs. Since if more of a help request than an bug report, I moving this to a discussion.

**Comment by Bazimek** [comment]
Yes sorry for putting it to the bug topic. I tried add a network to the virtual machine with direct attachment ->enp5s0 but still I cannot connect to it from external acess (port 5900 ... 5905) is ofc forwarded in my router.

Yesterday I tried create a brige in the network module for enp5s0, but my whole machine died and I cannot connect to it. I really do not know what happened and why but it was really stresfull situation.

I connect to my machine locally (not via SSH or something like that, just keyboard) and I restart the machine. It was took so long, but after that (I did not do anything) the connection was resumed and in the network manager log I saw something like:
<info>  [1644867124.2148] checkpoint[0x55d8bf07a2a0]: rollback of /org/freedesktop/NetworkManager/Checkpoint/1

So I guess there is something what can automatically reverse that? I really do not know what it was, but really it saved me. Every connection for every app was restored after that.

**Comment by guercheLE** [comment]
I have guacamole running as a podman container (10.89.0.* network). I have tried configuring guacamole to connect to 192.168.31.254 (host), but unsuccessfully. I guess it did not connect because VM (192.168.122.*) listen to and accepts only incoming connections from 127.0.0.1. I tried installing vnc and/or xrdp on the VMs, but guacamole container (10.89.0.* network) does not reach vnc/xrdp installed inside VM also (192.168.122.*).
